lib/msf/core/modules/loader/base.rb

```diff
@@ -468,7 +468,7 @@ def module_path?(path)
   def script_path?(path)
     # warn users if their external modules aren't marked executable
     # per #14281
-    if File.directory?(path) || !['#!', '//'].include?(File.read(path, 2))
+    if File.directory?(path) || !['#!', '//'].include?(File.read(path, 2)) || File::basename(path).start_with?('example')
```

Comment:
Instead of complicating this method more, we should add this check to the executable loader the same as we do for classic style modules: https://github.com/rapid7/metasploit-framework/blob/a890210bfc52b3dd3679ee05777507d97e1d6f8a/lib/msf/core/modules/loader/directory.rb#L45

Reply:
Given what is said here this would just mean changing the line at https://github.com/rapid7/metasploit-framework/blob/85a9accbeec063f52406621f124a7f82ad63dcba/lib/msf/core/modules/loader/executable.rb#L45 to what is shown above, which I think is fair.

Granted a question could be raised as to whether or not this check should take place earlier, but after speaking to Adam about this more, I do agree with his option that such work should really be reserved for a separate PR. For now we should focus on placing this check in its appropriate location in line with the check that on line 45 of `directory.rb`, as shown above, so that we can more easily refactor this code at a later date if needs be.

I'll go ahead and edit this so long 👍

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/14383/commits/7b7bdff1cc393fbfcf6180e1178905948f08318a